client/src/components/TODOList/TODOCalendarView.js

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from "react";
+import { Spinner } from "@material-tailwind/react";
+import Calendar from "react-calendar";
+import "react-calendar/dist/Calendar.css";
+import { useSelector, useDispatch } from "react-redux";
+import thunk from "../../store/TODOList/thunk";
+import TODOCalendarIcon from "./TODOCalendarIcon";
+
+// Renders a Calendar view of all the TODOItems created. Each TODOItem
+// is added to a Calendar "Date" tile based upon its startDate attribute.
+const TODOCalendarView = ({ selectedCategoryID }) => {
+  const { TODOList, fetchTODOList } = useSelector((state) => state.todoReducer);
+  const user = useSelector((state) => state.loginReducer);
+
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState(false);
+
+  // Fetch TODOList from Redux store whenever list is updated
+  useEffect(() => {
+    setLoading(true);
+    if (user.isLoggedIn) {
+      dispatch(
+        thunk.getTODOListAsync({
+          userID: user.user.userID,
+          categoryID: selectedCategoryID,
+        })
+      ).then(() => {
+        setLoading(false);
+      });
+    }
+  }, [dispatch, fetchTODOList, selectedCategoryID, user]);
+
+  // Determines which TODOItems belong to a given "date" cell based
+  // their "start date" attribute
+  const findTODOsThatBelongToGivenDate = (date) => {
+    return TODOList.filter((todo) => {
+      // convert "startDate" date string into Date object
+      const startDate = new Date(todo.startDate);
+
+      return (
+        startDate.getFullYear() === date.getFullYear() &&
+        startDate.getMonth() === date.getMonth() &&
+        startDate.getDate() === date.getDate()
+      );
+    });
+  };
+
+  // For items whose "startDate" attribute matches the
+  // current "date" tile, convert each of those items into TODOItem
+  // components and add them into the given "date" tile of the
+  // Calendar.
+  const populateDateTileWithTODOs = ({ date }) => {
+    const TODOsForGivenDate = findTODOsThatBelongToGivenDate(date);
+
+    // check whether there are more than zero TODOItems whose start date
+    // matches the current "Date" tile of the Calendar
+    const isNumTODOMoreThanZero = TODOsForGivenDate.length > 0;
+
+    return (
+      <>
+        {isNumTODOMoreThanZero && (
+          <div className="w-4">
+            <TODOCalendarIcon TODOList={TODOsForGivenDate} />
+          </div>
+        )}
+      </>
+    );
+  };
+
+  return (
+    <div className="flex flex-col items-center mt-4">
+      {loading && <Spinner className="h-10 w-10" />}
+      <Calendar tileContent={populateDateTileWithTODOs} />
```

Comment:
Is it intended that clicking on a date in the calendar (with or without TODOs for that day) causes the date number to disappear? Is it also intended that TODOs for that day only appear if you click on the red exclamation mark and not also the date number with the TODOs?:

![image](https://github.com/awctw/study-dash/assets/56493743/207b5bfe-97e4-4361-b61b-ccd1c2f7d867)
![image](https://github.com/awctw/study-dash/assets/56493743/9e683e22-bf22-4659-ad2b-3e1f7443ad33)
![image](https://github.com/awctw/study-dash/assets/56493743/5b70224b-909f-46e5-bda9-cbd65a7d2808)

Reply:
The white space that occurs when you click on a date tile is an issue that I was not able to fix.

As for clicking the icon to show the TODO items, yes that is intentional

Reply:
Perhaps look into the absence of the props listed in the [react-calendar](https://www.npmjs.com/package/react-calendar) page. For example, the simple example of its basic usage involves onChange and so excluding it may have unexpected side effects.

Reply:
That or look into the styling for selection (i.e. selection highlights in white instead of the default blue?)

Reply:
It is not a trivial issue. In the TODOList.css file, you can see me attempting to set the background colour of the selected tile to be transparent (not solid white). But it is not working. So fixing this might need some time.

Reply:
I assume this css is required for the red exclamation mark to show up or something but if you don't apply any of your own css to the component does the default selection of react-calendar still work with the numbers not being hidden?

Reply:
No. The TODOList.css contains the styles I have applied to each date tile of the Calendar. It does not directly affect the todo icon in a tile

Reply:
If I comment out the styles I applied to the Calendar tiles, the issue still exists

Reply:
react-calendar documentation is not too great

Reply:
Alright if this isn't easily resolvable by tomorrow I will approve for now.